pkg/virt-controller/watch/vmi.go

```diff
@@ -490,32 +490,24 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 }
 
 // isPodReady treats the pod as ready to be handed over to virt-handler, as soon as all pods except
-// the compute pod are ready. That includes kubevirt-infra and sidecars.
+// the compute pod are ready.
 func isPodReady(pod *k8sv1.Pod) bool {
 	if isPodDownOrGoingDown(pod) {
 		return false
 	}
-	infraContainerExists := false
-	for _, containerStatus := range pod.Status.ContainerStatuses {
-		if containerStatus.Name == "kubevirt-infra" {
-			infraContainerExists = true
-			break
-		}
-	}
+
 	for _, containerStatus := range pod.Status.ContainerStatuses {
 		// If there is a kubevirt-infra pod, we should not check "compute" here.
 		// Otherwise all containers including "compute" should be "ready before we go on.
-		if infraContainerExists {
-			if containerStatus.Name != "compute" && containerStatus.Ready == false {
+		if containerStatus.Name == "compute" {
```

Comment:
why is the compute container getting special treatment here still? how is the compute container different from any of the other containers now?

Reply:
The `compute` container will hold the readiness probes of the VMI if some are specified. The `ready` status can't be taken into account when handing the VMI over to virt-handler. We only check for this container that it actually started and wait for it to become reade when connecting in virt-handler.